project/Build.scala

```diff
@@ -751,9 +751,18 @@ object Build {
   def dottyCompilerSettings(implicit mode: Mode): sbt.Def.SettingsDefinition =
     if (mode == NonBootstrapped) nonBootstrapedDottyCompilerSettings else bootstrapedDottyCompilerSettings
 
-  lazy val `scala3-compiler` = project.in(file("compiler")).asDottyCompiler(NonBootstrapped)
-  lazy val `scala3-compiler-bootstrapped` = project.in(file("compiler")).asDottyCompiler(Bootstrapped)
-
+  lazy val `scala3-compiler` = project.in(file("compiler")).
+    asDottyCompiler(NonBootstrapped).
+    settings(
+      Compile / packageBin / packageOptions +=
+        Package.ManifestAttributes("Automatic-Module-Name" -> s"$dottyOrganization-${name.value}")
```

Comment:
According to https://docs.oracle.com/javase/specs/jls/se16/html/jls-7.html#jls-7.7:
> A module name consists of one or more Java identifiers (§3.8) separated by "." tokens. 

 A java identifier can't have a `-` in its name, so I don't think this is a valid name, use a `.` instead.

Reply:
Similarly, you probably need to replaceAll("-", ".") to handle the "-" in the project names.

Reply:
Also instead of name I think we should use moduleName, that's what shows up in the pom in the end.